src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+
+namespace System.IO.IsolatedStorage
+{
+    internal static partial class Helper
+    {
+        private const string IsolatedStorageDirectoryName = "IsolatedStorage";
+
+        private static string s_MachineRootDirectory;
+        private static string s_RoamingUserRootDirectory;
+        private static string s_UserRootDirectory;
+
+        private static readonly char[] s_Base32Char =
+        {
+            'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h',
+            'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p',
+            'q', 'r', 's', 't', 'u', 'v', 'w', 'x',
+            'y', 'z', '0', '1', '2', '3', '4', '5'
+        };
+
+        internal static string GetRootDirectory(IsolatedStorageScope scope)
+        {
+            // The full root directory is the relevant special folder from Environment.GetFolderPath() plus "IsolatedStorage"
+            // and a set of random directory names if not roaming. The current identity gets tacked on after this.
+
+            if (IsRoaming(scope))
+            {
+                if (string.IsNullOrEmpty(s_RoamingUserRootDirectory))
+                    s_RoamingUserRootDirectory = GetDataDirectory(scope);
```

Comment:
This is not a single line if, it is a nested if which are were most of bracing type errors end up occurring. This is called out directly in the guideline.